src/GovUk.Education.ExploreEducationStatistics.Admin/Models/Either.cs

```diff
@@ -62,6 +62,18 @@ public static class EitherTaskExtensions
             var next = await func();
             return new Either<Tl, T>(next);
         }
+        
+        public static async Task<Either<Tl, T>> OnSuccess<Tl, Tr, T>(this Task<Either<Tl, Tr>> task, Func<Tr, Task<Either<Tl, T>>> func)
```

Comment:
Might wanna see what the C# "promises" chaining naming convention is e.g. "Then()" might be a more well-used term than "OnSuccess()"

Reply:
I could do but the reason I've put it as OnSuccess is that it'll only ever be called if the first call is a success - which is a bit different from a promise.....